rrdtool/rrd_mts_similarity.c

```diff
@@ -363,63 +364,57 @@ int main(int argc, char *argv[]) {
 
   filename = malloc(sizeof(char*)*MAX_NUM_FILE);
   for(i=0; i<MAX_NUM_FILE; i++)
+  {
     filename[i] = malloc(sizeof(char)*50);
-
-  while((c = getopt(argc, argv, "d:s:e:a:qf:t:vz")) != '?') {
-    if(c == -1) break;
-
-    switch(c) {
-    case 's':
-      start_s = optarg;
-      break;
-
-    case 'd':
-      basedir = optarg;
-      break;
-
-    case 'e':
-      end_s = optarg;
-      break;
-
-    case 'q':
-      quick_mode = 1;
-      break;
-
-    case 'v':
-      verbose = 1;
-      break;
-
-    case 'a':
-      {
-	float f = atof(optarg);
-
-	if((f > 0) && (f < 1))
-	  alpha = f;
-	else
-	  printf("Discarding -a: valid range is >0 .. <1\n");
-      }
-      break;
-
-    case 'f':
-      filename[n_file] = optarg;
-      n_file++;
-      break;
-
-    case 't':
-      similarity_threshold = atoi(optarg);
-      break;
-
-    case 'z':
-      skip_zero = 1;
-      break;
-
-    default:
-      help();
-      break;
-    }
+    filename[i] = NULL;
+  }
+  
+  for(i= 0; i<argc;i++)
```

Comment:
Used for the parsing of the arguments instead of while((c = getopt(argc, argv, "d:s:e:a:qf:t:vz")) != '?'),  improved -f

Reply:
Not a good idea. Please restore it.

Reply:
Fixed in commit with ID 030a0c3